From 4cbc0688c5b1db7bd633542572690d6d61d5f343 Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Tue, 28 Apr 2015 10:10:58 +0100
Subject: [PATCH 1/2] regex test: do not assert that system PCRE still has an
 8.31 bug

This was fixed in 8.32, so if we have that version, assert that it is
fixed; if we don't (e.g. the current internal pcre), still don't
assert that it *isn't* fixed.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=733325
---
 glib/tests/regex.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/glib/tests/regex.c b/glib/tests/regex.c
index b886b06..5e5b88d 100644
--- a/glib/tests/regex.c
+++ b/glib/tests/regex.c
@@ -25,6 +25,12 @@
 #include <locale.h>
 #include "glib.h"
 
+#ifdef USE_SYSTEM_PCRE
+#include <pcre.h>
+#else
+#include "glib/pcre/pcre.h"
+#endif
+
 /* U+20AC EURO SIGN (symbol, currency) */
 #define EURO "\xe2\x82\xac"
 /* U+00E0 LATIN SMALL LETTER A WITH GRAVE (letter, lowercase) */
@@ -2446,8 +2452,12 @@ main (int argc, char *argv[])
   /* Test that othercasing in our pcre/glib integration is bug-for-bug compatible
    * with pcre's internal tables. Bug #678273 */
   TEST_MATCH("[Ǆ]", G_REGEX_CASELESS, 0, "Ǆ", -1, 0, 0, TRUE);
-  TEST_MATCH("[Ǆ]", G_REGEX_CASELESS, 0, "ǅ", -1, 0, 0, FALSE);
   TEST_MATCH("[Ǆ]", G_REGEX_CASELESS, 0, "ǆ", -1, 0, 0, TRUE);
+#if PCRE_MAJOR > 8 || (PCRE_MAJOR == 8 && PCRE_MINOR >= 32)
+  /* This would incorrectly fail to match in pcre < 8.32, so only assert
+   * this for known-good pcre. */
+  TEST_MATCH("[Ǆ]", G_REGEX_CASELESS, 0, "ǅ", -1, 0, 0, TRUE);
+#endif
 
   /* TEST_MATCH_NEXT#(pattern, string, string_len, start_position, ...) */
   TEST_MATCH_NEXT0("a", "x", -1, 0);
-- 
2.1.4

